Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -174,5 +175,24 @@ public async Task<CourseModificationRequest> InitiateCourseModification(CourseMo
         _logger.LogInformation($"Created ${typeof(CourseCreationRequest)} ${courseModificationRequest.Id}");
 
         return courseModificationRequest;
+    }
+
+    public async Task<Course?> GetCourseData(CourseDTO c)
+    {
+        var course = await _courseRepository.GetCourseBySubjectAndCatalog(c.Subject, c.Catalog);
+        if (course == null)
+        {
+            throw new ArgumentException("The course does not exist.");
+        }
+
+        var courseWithLatestVersion = await _courseRepository.GetCourseByCourseIdAndLatestVersion(course.CourseID);
+
+        if (courseWithLatestVersion == null || courseWithLatestVersion.CourseState == CourseStateEnum.Deleted)
+        {
+            throw new ArgumentException("The course is deleted.");
+        }
+
+        return await _courseRepository.GetCourseData(courseWithLatestVersion);

```

Comment:
What's this final call for? What does it provide that's not already in `courseWithLatestVersion`?

Reply:
You are right, i will simply return courseWithLatestVersion